src/main/java/gregtech/api/metatileentity/multiblock/ui/MultiblockUIBuilder.java

```diff
@@ -0,0 +1,984 @@
+        maintenanceProblems = getSyncer().syncByte(maintenanceProblems);
+        if (maintenanceProblems < 63) {
+            addKey(KeyUtil.lang(TextFormatting.YELLOW,
+                    "gregtech.multiblock.universal.has_problems"));
+
+            // Wrench
+            if ((maintenanceProblems & 1) == 0) {
+                addKey(KeyUtil.lang(TextFormatting.GRAY,
+                        "gregtech.multiblock.universal.problem.wrench"));
+            }
+
+            // Screwdriver
+            if (((maintenanceProblems >> 1) & 1) == 0) {
+                addKey(KeyUtil.lang(TextFormatting.GRAY,
+                        "gregtech.multiblock.universal.problem.screwdriver"));
+            }
+
+            // Soft Mallet
+            if (((maintenanceProblems >> 2) & 1) == 0) {
+                addKey(KeyUtil.lang(TextFormatting.GRAY,
+                        "gregtech.multiblock.universal.problem.soft_mallet"));
+            }
+
+            // Hammer
+            if (((maintenanceProblems >> 3) & 1) == 0) {
+                addKey(KeyUtil.lang(TextFormatting.GRAY,
+                        "gregtech.multiblock.universal.problem.hard_hammer"));
+            }
+
+            // Wire Cutters
+            if (((maintenanceProblems >> 4) & 1) == 0) {
+                addKey(KeyUtil.lang(TextFormatting.GRAY,
+                        "gregtech.multiblock.universal.problem.wire_cutter"));
+            }
+
+            // Crowbar
+            if (((maintenanceProblems >> 5) & 1) == 0) {
+                addKey(KeyUtil.lang(TextFormatting.GRAY,
+                        "gregtech.multiblock.universal.problem.crowbar"));
+            }
+        }
+        return this;
+    }
+
+    /**
+     * Adds two error lines when the machine's muffler hatch is obstructed.
+     * <br>
+     * Added if the structure is formed and if the passed parameter is true.
+     */
+    public MultiblockUIBuilder addMufflerObstructedLine(boolean isObstructed) {
+        if (!isStructureFormed) return this;
+        if (getSyncer().syncBoolean(isObstructed)) {
+            addKey(KeyUtil.lang(TextFormatting.RED,
+                    "gregtech.multiblock.universal.muffler_obstructed"));
+            addKey(KeyUtil.lang(TextFormatting.GRAY,
+                    "gregtech.multiblock.universal.muffler_obstructed_desc"));
+        }
+        return this;
+    }
+
+    /**
+     * Adds a fuel consumption line showing the fuel name and the number of ticks per recipe run.
+     * <br>
+     * Added if structure is formed, the machine is active, and the passed fuelName parameter is not null.
+     */
+    public MultiblockUIBuilder addFuelNeededLine(String fuelName, int previousRecipeDuration) {
+        if (!isStructureFormed || !isActive) return this;
+        fuelName = getSyncer().syncString(fuelName);
+        previousRecipeDuration = getSyncer().syncInt(previousRecipeDuration);
+        addKey(KeyUtil.lang(TextFormatting.GRAY,
```

Comment:
This isn't working correctly in the Combustion Engine. It displays the amount of fuel consumed, but not the name of the fluid.

Reply:
the parameter was misleading, it is supposed to be the amount as a string